swarm/pss/pss.go

```diff
@@ -538,37 +529,29 @@ func (self *Pss) send(to []byte, topic whisper.TopicType, msg []byte, pubkey *ec
 	if err != nil {
 		return fmt.Errorf("failed to generate whisper message encapsulation: %v", err)
 	}
-	// performs encryption and PoW
+	// performs encryption.
+	// Does NOT perform / performs negligible PoW due to very low difficulty setting
 	// after this the message is ready for sending
 	envelope, err := woutmsg.Wrap(wparams)
 	if err != nil {
 		return fmt.Errorf("failed to perform whisper encryption: %v", err)
 	}
-	log.Trace("pssmsg whisper done", "env", envelope, "wparams payloda", wparams.Payload)
+	log.Trace("pssmsg whisper done", "env", envelope, "wparams payload", wparams.Payload)
 	// prepare for devp2p transport
 	pssmsg := &PssMsg{
-		To:      to,
+		To:      to[:addresslength],
 		Payload: envelope,
 	}
 	return self.Forward(pssmsg)
 }
 
 // Forwards a pss message to the peer(s) closest to the to recipient address in the PssMsg struct
 //
-// The To-field in the PssMsg struct will be plaintext, unencrypted, and will be truncated to the magnitude set by SetRecipientAddressLength (PssParams.RecipientAddressLength by default).
-// If the recipientAddressLength value is 0, no bytes will be disclosed, and message will be forwarded to ALL devp2p peers
-//
 // Handlers that are merely passing on the PssMsg to its final recipient might call this directly
 func (self *Pss) Forward(msg *PssMsg) error {
-	to := msg.To
-	zeros := make([]byte, len(to)-int(self.recipientAddressLength))
-	// truncate part of the pssmsg recipient address
-	// this way noone can know for certain exactly who it's for
-	// the kademlia address will be same length
-	// but equivalent to truncated part will be overwritten by zeros
-	copy(to[self.recipientAddressLength:len(to)], zeros)
-	log.Trace("truncated recipient", "original", msg.To, "truncated", to)
-	msg.To = msg.To[:self.recipientAddressLength]
+	var potaddr pot.Address // most authoritative data type for determining length of full address
```

Comment:
partial addressing in bitlength but to address can only be bytes.
This tells me there is no way out of having to specify the address length for routing explicitly in the message struct:
beside `To` we need `MaxLum` to specify the luminous bits.
This param needs to be passed to `Forward` and `Process`